packages/lit-html/src/lib/lit-html.ts

```diff
@@ -233,12 +233,17 @@ class Template {
           // HTML, like `<?` or `</{`. Figure out whether / how to handle that.
           if (match[1] === '!--') {
             regex = commentRegex;
-          } else {
+          } else if (match[2] === undefined) {
+            // If match[2] !== undefined, then we have an unescaped < and stay
+            // in the text state
             if (rawTextElement.test(match[1])) {
-              regex = new RegExp(`<\/${match[1]}`);
+              regex = new RegExp(`<\/${match[1]}`, 'g');
```

Comment:
Adding on @jridgewell's earlier suggestion, we could also combine `rawTextElement` into `textRegex` pretty easily with another group to avoid running a second regex on every tag name.
```js
const textRegex = /<(!--|\/?((script|style|textarea)|\w+|$))/g;
```

Then the 4 cases are:
```js
if (match[1] === '!--') {
  // comment open
} else if (match[2] === '') {
  // dynamic tag open/close
} else if (match[3] !== undefined) {
  // raw text element
} else {
  // normal tag open/close
}

Reply:
I didn't really want to combine the `rawTextElement` regex because it's also used during the tree walk, and I changed it to be anchored at the beginning and end of the string.